### DISCUSSION TITLE
Add Hooks

### DISCUSSION THREAD

**Discussion by joshmedeski** [pwntester/octo.nvim#472] [Ideas]
I'd love to see hooks implemented in Octo.

As an example, when I open [diffview.nvim](https://github.com/sindrets/diffview.nvim) I want to decrease the font size of my terminal emulator (see [my dotfiles](https://github.com/joshmedeski/dotfiles/blob/2fbffbe345551ed535718dc0292fca6e48f8004b/.config/nvim/lua/plugins/diffview.lua#L20)).

Could we add hook functionality to Octo? Perhaps a hook for pull requests:
```lua
require"octo".setup({
  hooks = {
    review_view_opened = function()
    end
    review_view_closed = function()
    end
    issue_view_opened = function()
    end
    issue_view_closed = function()
    end
    pr_view_opened = function()
    end
    pr_view_closed = function()
    end
  }
})
```

I'd be willing to help with development if you think this is a good idea worth moving forward with @pwntester

**Comment by pwntester** [comment]
Sounds great to me 👍🏼
Let me know if you need any info or context
